src/config.cc

```diff
@@ -78,127 +145,85 @@ namespace detail {
 			MPI_Comm_free(&host_comm);
 		}
 
-		// ------------------------------- CELERITY_LOG_LEVEL ---------------------------------
+		auto pref = env::prefix("CELERITY");
+		const auto env_log_level = pref.register_option<log_level>(
+		    "LOG_LEVEL", {log_level::trace, log_level::debug, log_level::info, log_level::warn, log_level::err, log_level::critical, log_level::off});
+		const auto env_gpmv = pref.register_variable<size_t>("GRAPH_PRINT_MAX_VERTS");
+		const auto env_devs = pref.register_variable<std::vector<size_t>>("DEVICES");
+		const auto env_profile_ocl = pref.register_variable<int>("PROFILE_OCL");
+		const auto env_profile_kernel = pref.register_variable<int>("PROFILE_KERNEL");
+		const auto env_force_wg = pref.register_variable<bool>("FORCE_WG");
+		const auto env_dry_run_nodes = pref.register_variable<size_t>("DRY_RUN_NODES");
+
+		const auto parsed_and_validated_envs = pref.parse_and_validate();
+		if(parsed_and_validated_envs.ok()) {
```

Comment:
Most if not all of the logic below should probably be a validation step of libenvpp, as that is what validations were designed for. The idea is that if there's a value that can be retrieved from libenvpp after parsing and validating that value can directly be used, without needing further checking.